Makefile.am

```diff
@@ -65,29 +65,19 @@ endif
 @DX_RULES@
 MOSTLYCLEANFILES += $(DX_CLEANFILES)
 
-### Library ###
-lib_LTLIBRARIES += libtpm2-totp.la
-include_HEADERS += include/tpm2-totp.h
-
-libtpm2_totp_la_SOURCES = src/libtpm2-totp.c
-libtpm2_totp_la_LIBADD = $(AM_LDADD)
-libtpm2_totp_la_LDFLAGS = $(AM_LDFLAGS) '(tpm2_totp)'
-
-pkgconfig_DATA = dist/tpm2-totp.pc
-
 ### Executable ###
 bin_PROGRAMS += tpm2-totp
 
-tpm2_totp_SOURCES = src/tpm2-totp.c
+tpm2_totp_SOURCES = src/tpm2-totp.c src/libtpm2-totp.c
 tpm2_totp_CFLAGS = $(AM_CFLAGS) $(TSS2_TCTILDR_CFLAGS) $(QRENCODE_CFLAGS)
-tpm2_totp_LDADD = $(AM_LDADD) $(TSS2_TCTILDR_LIBS) $(QRENCODE_LIBS) -ldl libtpm2-totp.la
+tpm2_totp_LDADD = $(AM_LDADD) $(TSS2_TCTILDR_LIBS) $(QRENCODE_LIBS)
 tpm2_totp_LDFLAGS = $(AM_LDFLAGS)
 
 if HAVE_PLYMOUTH
 helpers_PROGRAMS += plymouth-tpm2-totp
-plymouth_tpm2_totp_SOURCES = src/plymouth-tpm2-totp.c
+plymouth_tpm2_totp_SOURCES = src/plymouth-tpm2-totp.c src/libtpm2-totp.c
 plymouth_tpm2_totp_CFLAGS = $(AM_CFLAGS) $(TSS2_TCTILDR_CFLAGS) $(PLY_BOOT_CLIENT_CFLAGS)
-plymouth_tpm2_totp_LDADD = $(AM_LDADD) $(TSS2_TCTILDR_LIBS) $(PLY_BOOT_CLIENT_LIBS) -ldl libtpm2-totp.la
+plymouth_tpm2_totp_LDADD = $(AM_LDADD) $(TSS2_TCTILDR_LIBS) $(PLY_BOOT_CLIENT_LIBS)
```

Comment:
Could you provide a use case for this change, please? In my opinion, inlining only libtpm2-totp is not particularly useful: it removes the dependency on a single dynamic library, while the `tpm2-totp` executable still pulls in libqrencode and the TPM2 TSS stack (libtss2-esys, libtss2-mu, libtss2-tctildr) that you will have to deal with anyway.

If you want to avoid dynamic libraries as much as possible, I suggest looking into static linking instead, which can be achieved using
```
./configure LDFLAGS='-static' PKG_CONFIG='pkg-config --static'
```
This will remove the dependency on libtpm2-totp as well as the other libraries, provided static libraries for these are available on your system.

Reply:
Using `LDFLAGS=-static` links in *all* of the executables, as you pointed out, which makes for a much larger binary.

Additionally, all of the other dependencies can be met by system packages, so with `libtpm2_totp.c` linked into the executable it no longer requires any additional files or `LD_LIBRARY_PATH` configuration to run. If there were a debian package for `tpm2_totp` that handled installing the libraries into the normal places it wouldn't be too bad, but for building from source it requires extra setup or wrapper scripts.

Reply:
Ah ok, I was thinking more about an extremely minimal initramfs or embedded device where full static linking might make sense or even be mandatory.

Could you describe the steps you are using to build and install tpm2-totp, please? If you are using `make install`, libtool will automatically set the [rpath](https://en.wikipedia.org/wiki/Rpath) of installed executables to make sure that dynamic libraries it installs is found e.g. in `/usr/local/lib`. When you are just using tpm2-totp straight from the build directory, libtool creates a wrapper executable that sets the necessary environment variables. In both cases, no additional setup should be required to use tpm2-totp even if libtpm2-totp is not installed to `/usr/lib`.

Reply:
Not sure why static linking would make more sense in a minimal initramfs unless packages are unnecessarily bloated.  On the contrary, dynamic linking should make initramfs images smaller, which is desirable.